stores/moderationSubmissionsListStore.ts

```diff
@@ -0,0 +1,249 @@
+import { gql } from 'graphql-request'
+import { defineStore } from 'pinia'
+import { ref, type Ref } from 'vue'
+import { gqlClient } from '../utils/graphql.js'
+import type { Submission, Facility, HealthcareProfessional } from '~/typedefs/gqlTypes.js'
+
+export enum SelectedSubmissionListViewTab {
+    ForReview = 'FOR_REVIEW',
+    Approved = 'APPROVED',
+    Rejected = 'REJECTED'
+}
+
+export enum SelectedModerationListView {
+    Facilities = 'FACILITIES',
+    HealthcareProfessionals = 'HEALTHCARE_PROFESSIONALS',
+    Submissions = 'SUBMISSIONS'
+}
+
+export const useModerationSubmissionsListStore = defineStore(
+    'submissionsListStore',
+    () => {
+        const submissionsData: Ref<Submission[]> = ref([])
+        const healthcareProfessionalsData: Ref<HealthcareProfessional[]>
+        = ref([])
+        const facilityData: Ref<Facility[]> = ref([])
+        const selectedModerationListViewChosen: Ref<SelectedModerationListView> = ref(SelectedModerationListView.Submissions)
+        const selectedSubmissionId: Ref<string> = ref('')
+        const selectedFacilityId: Ref<string> = ref('')
+        const selectedHealthcareProfessionalId: Ref<string> = ref('')
+        const selectedSubmissionData: Ref<Submission | undefined> = ref()
+        const filteredSubmissionDataForListComponent: Ref<Submission[]> = ref([])
+
+        async function getSubmissions() {
+            const submissionsSearchResults = await querySubmissions()
+            submissionsData.value = submissionsSearchResults
+        }
+
+        async function getFacilities() {
+            const facilityResults = await queryFacilities()
+            facilityData.value = facilityResults
+        }
+
+        async function getHealthcareProfessionals() {
+            const healthcareProfessionalResults = await queryHealthcareProfessionals()
+            healthcareProfessionalsData.value = healthcareProfessionalResults
+        }
+
+        function setSelectedModerationListViewChosen(selectedOption: SelectedModerationListView) {
+            selectedModerationListViewChosen.value = selectedOption
+        }
+
+        function filterSelectedSubmission(submissionId: string | undefined) {
+            selectedSubmissionData.value = submissionsData.value.find(submission => submission.id === submissionId)
+        }
+
+        function filterSubmissionsBySelectedTab(tabValue: SelectedSubmissionListViewTab) {
```

Comment:
should keep this store data separate from the UI tab implementation. You can change this to filter by status and use the status enum (approved, rejected, inreview)